docs/dev/add-new-component.md

```diff
@@ -0,0 +1,48 @@
+# Adding a new component to our Collectors
+
+These instructions are for adding a new component to one of our collector distributions.
+
+## `distrogen` registries
+
+`distrogen` uses a registry yaml file to know how to refer to different components for distribution generation. In this repository there are 3 registry files:
+
+* [The embedded registry][embedded registry], which is embedded in the `distrogen` binary and contains all upstream `opentelemetry-collector` and `opentelemetry-collector-contrib` components
+* [`google-built-opentelemetry-collector` components local registry](../../components/google-built-opentelemetry-collector/registry.yaml) which contains components from this repo that are meant for the `google-built-opentelemetry-collector` distribution
+* [`otelopscol` components local registry](../../components/otelopscol/registry.yaml) which contains components from this repo that are meant for the `otelopscol` distribution
+
+These files contain components that can be referred to in `distrogen` spec files.
+
+## Adding upstream component
+
+To add a new upstream component to one of our collectors, you can refer to it in that collector's spec file (found in the `specs` folder).
+
+```yaml
+components:
+  receivers:
+  - componentname
+```
+
+The component name is the name of the component that is in the [`distrogen` embedded registry][embedded registry]. 
```

Comment:
"[`distrogen` embedded registry][embedded registry]" Can you fix the link reference?

Reply:
Fixed in [8b4ca94](https://github.com/GoogleCloudPlatform/opentelemetry-operations-collector/pull/306/commits/8b4ca94eef0cb8a5dd0a7d0a2b2bef422acd8f26)